### ISSUE TITLE
MD036 triggered right before a header

### ISSUE THREAD

**Issue by fharper** [DavidAnson/markdownlint#1026]
In this example

```
# Title

I have some text here.

_Note: something here_

## Another section

Other text here

```

The italic note will trigger the MD036 rule. I think it should not be triggered if there is a title immediately after, which would mean with no doubt that the emphasis isn't used as a title, since there is no parapgrah of text after it. WDYT?

**Comment by DavidAnson** [comment]
It's valid to have a heading that isn't followed by paragraph text – perhaps there is another heading or a block quote or the end of the file, etc.. I do not think the heuristic proposed is super reliable, and I'm reluctant to try to put too much intelligence into this rule. I think it is correct to warn in this case and clear why it did so. If this is a pattern you use frequently, you may want to disable the rule for that file or project.
